### ISSUE TITLE
My nll_loss is always 0 during training. Is this normal? The nll_loss corresponds to the LER loss from the paper？

### ISSUE THREAD

**Issue by gangxu822** [Cliffia123/LegoGCD#1]

**Comment by gangxu822** [comment]
![screenshot-20241024-174054](https://github.com/user-attachments/assets/e8e18eb9-a6fd-44b7-8b39-3c448bcdfbc0)

**Comment by Cliffia123** [comment]
I'm sorry for not getting back to you sooner. The `nll_loss` corresponds to the LER module in our method, with its implementation detailed as follows:  

`pseudo_label = torch.softmax((student_out/0.05), dim=-1)
delta_logits = torch.log(qhat)
logits_u = student_out/0.05 + 0.4*delta_logits
log_pred = F.log_softmax(logits_u, dim=-1)
nll_loss = torch.sum(-pseudo_label*log_pred, dim=1)*mask_old
`


If `nll_loss` evaluates to 0, it is advisable to verify whether `nll_loss` is set to False. If this condition holds, it suggests that the count of **potentially high-confidence known** samples is **zero**. In such cases, the threshold parameter `args.thr` can be adjusted to modify the number of samples classified as potentially high-confidence known samples.
